docs/site/tutorials/protocol_oriented_generics.ipynb

```diff
@@ -0,0 +1,1141 @@
+{
```

Comment:
<p>Why is it important for the reader to know that protocols are pervasive in S4TF? Will this knowledge (in comparison with the knowledge that you can extend something called Layer whatever this is) help them develop better models? Or the goal is to make the reader feel good about S4TF because it uses advanced engineering techniques such as protocol-oriented programming?</p>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/swift/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>